tests/testthat/helper_learners_all.R

```diff
@@ -34,3 +38,79 @@ testThatLearnerRespectsWeights = function(lrn, task, train.inds, test.inds, weig
   expect_true(any(get.pred.fun(p1) != get.pred.fun(p3)), info = lrn$id)
 }
 
+
+# Test that learner produces output on the console and whether a performance
+# measure is calculated which corresponds representing if learner works as it 
+# should.
+# This function is being used to test learners in general and in the other
+# helper functions testing learners that claim to handle missings, factors,...
+# It also tests if the learner can predict probabilities or standard errors.
+# When testing standard errors an additional test if there are as many predictions
+# as there are observations in the task is being performed.
+# Note: performance() needs the task argument so that it works with cluster learners.
+# further args: pred.type (only needs specification "prob" when testing learner
+# can predict probabilities or specification "se" when testing learner can
+# predict standard errors.)
+
+testThatLearnerCanTrainPredict = function(lrn, task, hyperpars, pred.type = "response") {
+  
+  if (lrn$id %in% names(hyperpars))
+    lrn = setHyperPars(lrn, par.vals = hyperpars[[lrn$id]])
+  
+  if (pred.type == "response") {
+    lrn = setPredictType.Learner(lrn, "response")
+  } else if (pred.type == "prob") {
+    lrn = setPredictType.Learner(lrn, "prob")
+  } else if (pred.type == "se") {
+    lrn = setPredictType.Learner(lrn, "se")
+  }
```

Comment:
The if statement seems unnecessary here. This could be just `lrn = setPredictType(lrn, pred.type)`.